### PR TITLE
fix: Load user details in Edit Profile section

### LINKED ISSUE #568: Load user details in Edit Profile section
When user open edit profile section and update profile picture, two Toast messages are display -
1. Please provide first name and last name
2. Image uploaded successfully
**While user first name and last name already registered on the server.**

Here progress bar is also hiding by soft keyboard.

### PR DESCRIPTION
**Pull request by liveHarshit**
Fixes #568

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt [hunk 1]
```diff
@@ -69,6 +82,10 @@ class EditProfileFragment : Fragment() {
         return rootView
     }
 
+    private fun hideSoftKeyboard() {
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: You can add this method in Utils.kt

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: > You can add this method in Utils.kt

Updated :+1: . Please review.

**Comment by liveHarshit** [comment]
Updated

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt [hunk 2]
```diff
@@ -50,7 +49,7 @@ class LoginFragment : Fragment() {
 
         rootView.loginButton.setOnClickListener {
             loginActivityViewModel.login(email.text.toString(), password.text.toString())
-            hideSoftKeyboard()
+            hideSoftKeyboard(rootView.context, rootView)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Don't pass view's context here

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt [hunk 3]
```diff
@@ -59,6 +70,7 @@ class EditProfileFragment : Fragment() {
         }
 
         rootView.buttonUpdate.setOnClickListener {
+            hideSoftKeyboard(rootView.context, rootView)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: Same here

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 4]
```diff
@@ -63,11 +62,11 @@ class ProfileFragment : Fragment() {
 
         profileFragmentViewModel.user.observe(this, Observer {
             it?.let {
-                rootView.name.text = "${it.firstName.nullToEmpty()} ${it.lastName.nullToEmpty()}"
+                val userName = "${it.firstName.nullToEmpty()} ${it.lastName.nullToEmpty()}"
+                rootView.name.text = userName
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: Why this change?

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 5]
```diff
@@ -82,7 +81,7 @@ class ProfileFragment : Fragment() {
     }
 
     override fun onOptionsItemSelected(item: MenuItem?): Boolean {
-        when (item?.getItemId()) {
+        when (item?.itemId) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: Why this change?

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 6]
```diff
@@ -63,11 +62,11 @@ class ProfileFragment : Fragment() {
 
         profileFragmentViewModel.user.observe(this, Observer {
             it?.let {
-                rootView.name.text = "${it.firstName.nullToEmpty()} ${it.lastName.nullToEmpty()}"
+                val userName = "${it.firstName.nullToEmpty()} ${it.lastName.nullToEmpty()}"
+                rootView.name.text = userName
                 rootView.email.text = it.email
                 emailSettings = it.email
 
-
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: Why this change?

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt [hunk 7]
```diff
@@ -32,9 +32,9 @@ class ProfileFragmentViewModel(private val authService: AuthService) : ViewModel
         compositeDisposable.add(authService.getProfile()
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
-                .doOnSubscribe({
+                .doOnSubscribe {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt]: WTC?

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt [hunk 8]
```diff
@@ -32,9 +32,9 @@ class ProfileFragmentViewModel(private val authService: AuthService) : ViewModel
...
                     progress.value = true
-                }).doFinally {
+                }.doFinally {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt]: WTC

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 9]
```diff
@@ -38,4 +38,9 @@ object Utils {
                .setPositiveButton(context?.resources?.getString(R.string.ok)) { dialog, _ -> dialog.cancel() }
         builder.show()
     }
+
+    fun hideSoftKeyboard(activity: Context, view: View) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: This is context, not activity

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt [hunk 10]
```diff
@@ -69,7 +81,6 @@ class EditProfileFragment : Fragment() {
         return rootView
     }
 
-
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: WTC

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt (hunk 4 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: > Why this change?

It shows warning so I pass it as String. Should I recover it?

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt (hunk 6 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: Extra feed. Should I recover it? 

> Why this change?

Extra feed. Should I recover it?

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt (hunk 7 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt]: > WTC?

Warning. Remove unnecessary parentheses as I mention in PR description. Should I recover it?

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt (hunk 8 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt]: > WTC

Same as above.

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt (hunk 10 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: Extra feed. Should I recover it?

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: > WTC

Extra feed. Should I recover it?

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt (hunk 5 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: > Why this change?

It automatically converts. But after check commit different when I try to edit it, studio automatic input itemId instead of getItemId(). Should I recover it?

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt (hunk 4 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: What warning does it show? Ideally, we only fix issue related things in one PR

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt (hunk 5 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: OK, let it be

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt (hunk 6 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: Yes

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt (hunk 7 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt]: Yes

**Comment by iamareebjamal** [comment]
Yes, please revert all unrelated changes from the PR. You can send a new PR to handle warnings

**Comment by liveHarshit** [comment]
> Yes, please revert all unrelated changes from the PR. You can send a new PR to handle warnings

Okay got it. By next time I will change in code related to the issue only. For extra changes or warning, need to submit new PR related to new issue. :+1:

**Comment by liveHarshit** [comment]
updated

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt [hunk 11]
```diff
@@ -59,6 +70,7 @@ class EditProfileFragment : Fragment() {
         }
 
         rootView.buttonUpdate.setOnClickListener {
+            hideSoftKeyboard(context!!.applicationContext, rootView)
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: Wrong, just pass context, no need to do unsafe cast and get application context

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt (hunk 5 above)
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: I thought this file was part of change, since this file is not a part of changes, please revert this

**Comment by liveHarshit** [comment]
Updated

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 12]
```diff
@@ -38,4 +38,9 @@ object Utils {
                .setPositiveButton(context?.resources?.getString(R.string.ok)) { dialog, _ -> dialog.cancel() }
         builder.show()
     }
+
+    fun hideSoftKeyboard(context: Context, view: View) {
+        val inputManager: InputMethodManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
```
[the remark this answers is not in the document]
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: @iamareebjamal It still showing Unsafe cast. Please suggest me what should I do.

[the remark this answers is not in the document]
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: @iamareebjamal It still showing Unsafe cast. Please suggest me what should I do.

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Let it be

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt [hunk 13]
```diff
@@ -59,6 +70,7 @@ class EditProfileFragment : Fragment() {
         }
 
         rootView.buttonUpdate.setOnClickListener {
+            hideSoftKeyboard(context!!, rootView)
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: Remove `!!`

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt (hunk 12 above)
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Make context nullable

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 14]
```diff
@@ -38,4 +38,9 @@ object Utils {
                .setPositiveButton(context?.resources?.getString(R.string.ok)) { dialog, _ -> dialog.cancel() }
         builder.show()
     }
+
+    fun hideSoftKeyboard(context: Context?, view: View) {
+        val inputManager: InputMethodManager = context?.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
```
[the remark this answers is not in the document]
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: @iamareebjamal Please review.
